src/base.js

```diff
@@ -320,9 +340,13 @@ export default class KintoClientBase {
   @nobatch("Can't use batch within a batch!")
   async batch(fn, options = {}) {
     const rootBatch = new KintoClientBase(this.remote, {
-      ...this._options,
-      ...this._getRequestOptions(options),
+      events: this.events,
       batch: true,
+      safe: this._getSafe(options),
+      // FIXME: this doesn't actually matter, probably, since it gets
+      // passed as "default headers" in the batch?
+      headers: this._getHeaders(options),
```

Comment:
The only case where I think this could matter is if we ever combined headers (for example, turning `Content-Disposition: attachment` into `Content-Disposition: attachment; filename="filename.jpg"`. But our headers in Kinto never do that; we only ever add new headers, or replace existing ones. So semantically, leaving them in `defaults` should be the same as using them in all the requests, plus it leads to a smaller request. No tests break when I remove this, so I'm going to take it out.